temporal-sdk/src/main/java/io/temporal/internal/worker/PollTaskExecutor.java

```diff
@@ -74,18 +75,7 @@ public interface TaskHandler<TT> {
               });
     } else {
       ThreadPoolExecutor threadPoolTaskExecutor =
-          new ThreadPoolExecutor(
-              // for SynchronousQueue we can afford to set it to 0, because the queue is always full
-              // or empty
-              // for LinkedBlockingQueue we have to set slots to workerTaskSlots to avoid situation
-              // when the queue grows, but the amount of threads is not, because the queue is not
-              // (and
-              // never) full
-              synchronousQueue ? 0 : workerTaskSlots,
-              workerTaskSlots,
-              10,
-              TimeUnit.SECONDS,
-              synchronousQueue ? new SynchronousQueue<>() : new LinkedBlockingQueue<>());
+          new ThreadPoolExecutor(0, threadPoolMax, 10, TimeUnit.SECONDS, new SynchronousQueue<>());
```

Comment:
So I am fine with this change, but it will mean that any submission to the `threadPoolTaskExecutor` will not be blocking if the `threadPoolMax` is full. Did you confirm the submitter can handle this and we have appropriate test coverage?

Reply:
Made sure this works via a small change to `testParallelLocalActivities`